tests/php/ModelAdminTest.php

```diff
@@ -38,14 +38,33 @@ public function testModelAdminOpens()
         $this->assertEquals(200, $this->get('ContactAdmin')->getStatusCode());
     }
 
+
+
+
     public function testMultiModelAdminOpensEachTab()
     {
         $admin = new MultiModelAdmin();
+        $this->logInWithPermission();
         foreach ($admin->getManagedModelTabs()->toNestedArray() as $tab) {
             $request = new HTTPRequest('GET', $tab['Link']);
             $request->setRouteParams(['ModelClass' => substr($tab['Link'], strlen('admin/multi/'))]);
+            $request->setSession(new Session([]));
             $admin->setRequest($request);
             $admin->doInit();
+            $this->assertEquals(
```

Comment:
When writing tests, it's not enough that a bit of code doesn't throw an exception. It could be doing something completely unexpected without throwing an exception.

It's also difficult afterwards to understand what the test is trying to accomplish.

Reply:
Yep fair enough, threw it together too quickly. Odd that this particular test requires the admin login and some others don't - nice work debugging.